src/test/test-udev.c

```diff
@@ -61,14 +61,17 @@ int main(int argc, char *argv[]) {
 
         test_setup_logging(LOG_INFO);
 
-        if (argc != 3) {
-                log_error("This program needs two arguments, %d given", argc - 1);
+        if (argc != 2 && argc != 3) {
+                log_error("This program needs one or two arguments, %d given", argc - 1);
                 return EXIT_FAILURE;
         }
 
         if (fake_filesystems() < 0)
                 return EXIT_FAILURE;
 
+        if (argc == 2)
+                return EXIT_SUCCESS;
+
```

Comment:
In this case, the value of `argv[1]` is not used...

How about accepting argc==3 or argc==1? If no argument is specified, then it only checks fake_filesystem.

Reply:
I think it should be impossible to run the test with no arguments and get a zero exit code because it makes it much easier to accidentally include it in the test suite run with `ninja -C ./build test`. I'd prefer to leave it as is. Or I could add something like `&& streq(argv[1], "check")` :-)

Reply:
#10734.